test/functional/insert.test.js

```diff
@@ -1891,9 +1889,14 @@ describe('Insert', function () {
               .batchSize(2)
               .toArray(function (err, docs) {
                 expect(err).to.not.exist;
-                var doc = docs.pop();
 
-                test.ok(doc.a._bsontype === 'Long');
+                docs.forEach((d, i) => {
+                  expect(d.a, `Failed on the document at index ${i}`).to.not.be.a('number');
+                  expect(d.a, `Failed on the document at index ${i}`).to.have.property('_bsontype');
+                  expect(d.a._bsontype, `Failed on the document at index ${i}`).to.be.equal('Long');
+                });
+
+                // expect(doc.a).to.have.property('_bsontype').to.be.equal('Long');
```

Comment:
Can you explain why these changes were necessary? Is this still testing the same thing?

Reply:
The bsonOptions weren't being inherited properly by the cursor, the first document would respect the options because it received them from the options parameter in the cursor's constructor. But subsequent getMores would get the bsonOptions off the operation 

https://github.com/mongodb/node-mongodb-native/blob/master/src/cursor/core_cursor.ts#L588

and this was just undefined and so we would fill in defaults that were different, so the following documents would have different deserialization. 

Anyway I changed the test to check that all of the documents `a` properties remain Long class objects rather than number primitives instead of just the first.